### PR TITLE
feat(debugger): adds scaled drawImage to debugger

### PR DESCRIPTION
**Pull request by 2xAA**
Adds a resizeObserver to the debugger and a new Vuex store action and
mutation to allow for more performant drawing to the debugger canvas.

### CODE REVIEW

File: src/components/CanvasDebugger.vue [hunk 1]
```diff
@@ -63,18 +86,16 @@ export default {
...
-      {
+      this.$modV.$worker.postMessage({
         type: "dispatch",
-        identifier: "outputs/setDebugContext",
-        payload: offscreen
-      },
-      [offscreen]
-    );
+        identifier: "outputs/resizeDebug",
+        payload: { width }
```
Comment (TimPietrusky) [inline src/components/CanvasDebugger.vue:96]: Why are you only sending over the `width`?

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
I have just one question.

File: src/components/CanvasDebugger.vue (hunk 1 above)
Comment (2xAA) [reply src/components/CanvasDebugger.vue:96]: The height is calculated from the aspect ratio of the output canvas as we want the preview canvas to always match the output canvas aspect ratio but to be the smallest size to keep the drawImage operations fast.

Comment (2xAA) [reply src/components/CanvasDebugger.vue:96]: See https://github.com/vcync/modv-3/pull/194/files#diff-ea45b26f7094f0c2672f06445d1afd06R293-R295 for the height calculation.

**Pull request closed without merging**